oxo.py

```diff
@@ -1,33 +1,99 @@
 class OxoBoard:
-    def __init__(self):
-        """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+    """
+        Number representations:
+            0 -> Empty
+            1 -> O
+            2 -> X
+
+        board -> Playable positions
+        board_width, board_height -> Board size
+        strike_combinations -> Each position for strike
+    """
+    board = {}
+
+    board_width = 3
+    board_height = 3
+
+    strike_combinations = [
+        [(0, 0), (1, 0), (2, 0)],
+        [(0, 1), (1, 1), (2, 1)],
+        [(0, 2), (1, 2), (2, 2)],
+
+        [(0, 0), (0, 1), (0, 2)],
+        [(1, 0), (1, 1), (1, 2)],
+        [(2, 0), (2, 1), (2, 2)],
+
+        [(0, 0), (1, 1), (2, 2)],
+        [(2, 0), (1, 1), (0, 2)],
+    ]
+
+    def __init__(self, width = None, height = None, strikes = None):
+        # Allow the board to be edited on creation
+        self.board_width = width or self.board_width
+        self.board_height = height or self.board_height
+        self.strike_combinations = strikes or self.strike_combinations
+
+        # Move counter
+        self.total_moves = self.board_width * self.board_height
+        self.current_moves = 0
+
+        #Generate an empty board
+        for x in xrange(self.board_width):
+            for y in xrange(self.board_height):
+                self.board[x, y] = 0
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        return self.board[x, y]
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        if self.board[x, y] == 0:
+            self.board[x, y] = mark
+            self.current_moves += 1
+            return True
+        else:
+            return False
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        return self.current_moves >= self.total_moves
```

Comment:
Interesting approach but it should work! Though could just be `==` -- if `self.current_moves > self.total_moves` then presumably something has gone wrong